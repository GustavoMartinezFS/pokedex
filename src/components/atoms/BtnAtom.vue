<template>
  <button
    :class="[
      'btn',
      disabled ? 'btn--disabled' : '',
      icon ? 'btn--with-icon' : 'btn--no-icon',
      `btn--${color}`
    ]"
    :style="{
      width: widthBt,
      maxWidth: `${maxWidthBt}px`,
      minWidth: `${minWidthBt}px`,
      height: `${height}px`,
      padding: `${padding}px`,
      margin: `${margin}px`,
      backgroundColor: isPressed ? pressedColor : active? color : disabledColor
    }"
    @click="handleClick"
    @mousedown="handleMouseDown"
    @mouseup="handleMouseUp"
    @mouseleave="handleMouseUp"
    :disabled="disabled"
  >
    <template v-if="icon">
      <span class="icon">
        <img :src="selectedIcon" alt="icon" />
      </span>
    </template>
    <span class="text">{{ name }}</span>
  </button>
</template>

<script lang="ts" setup>
import { defineProps, computed, ref } from 'vue';

const props = defineProps({
  color: {
    type: String,
    default: '#F22539',
  },
  disabledColor: {
    type: String,
    default: '#BFBFBF',
  },
  pressedColor: {
    type: String,
    default: '#C00E20',
  },
  active: {
    type: Boolean,
    default: false,
  },
  pressed: {
    type: Boolean,
    default: false,
  },
  disabled: {
    type: Boolean,
    default: false,
  },
  widthBt: {
    type: String,
    default: '100%',
  },
  minWidthBt: {
    type: Number,
    default: 131,
  },
  maxWidthBt: {
    type: Number,
    default: 275,
  },
  height: {
    type: Number,
    default: 44,
  },
  padding: {
    type: Number,
    default: 10,
  },
  margin: {
    type: String,
    default: '0',
  },
  name: {
    type: String,
    required: true,
  },
  icon: {
    type: String,
    default: null,
  },
});

const iconMap = {
  star: new URL('@/assets/star.svg', import.meta.url).href,
  list: new URL('@/assets/list.svg', import.meta.url).href,
};

const isPressed = ref(false);

const emit = defineEmits(['manage:click']);

const handleClick = () => {
  if (!props.disabled) {
    emit('manage:click');
  }
};

const handleMouseDown = () => {
  if (props.pressed) {
    isPressed.value = true;
  }
};

const handleMouseUp = () => {
  if (props.pressed) {
    isPressed.value = false;
  }
};

const selectedIcon = computed(() => iconMap[props.icon] || '');
</script>

<style scoped>
.btn {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  border-radius: 50px;
  cursor: pointer;
  transition: background-color 0.3s ease;
  box-sizing: border-box;
  border: none;
}

.btn--disabled {
  background-color: #d3d3d3;
  color: #a9a9a9;
  cursor: not-allowed;
}

.icon {
  margin-right: 10px;
}

.text {
  display: inline-block;
  font: 18px "Lato",sans-serif;
  color: var(--vt-c-white);
}
</style>
