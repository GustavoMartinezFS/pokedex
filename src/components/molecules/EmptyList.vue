<template>
  <div class="empty-list">
    <h1>Uh-oh!</h1>
    <p>You look lost on your journey!</p>
    <btn-atom @manage:click="handleClick" name="Go back home" widthBt="155" :active="true"/>
  </div>
</template>

<script lang="ts" setup>
import BtnAtom from "@/components/atoms/BtnAtom.vue";

const emit = defineEmits(['manage:goBack']);

const handleClick = () => {
  emit('manage:goBack');
};
</script>

<style scoped>
.empty-list {
  width: 100%;
  text-align: center;
  padding: 20px;
}

h1 {
  color: #333;
  font: bold 36px "Lato", sans-serif;
}

p {
  font: 20px "Lato", sans-serif;
  font-weight: 500;
  padding: 10px 0 30px 0;
}
</style>
