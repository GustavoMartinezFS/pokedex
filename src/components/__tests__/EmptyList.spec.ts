import { mount } from '@vue/test-utils';
import EmptyList from '@/components/molecules/EmptyList.vue';
import BtnAtom from '@/components/atoms/BtnAtom.vue';

describe('EmptyList.vue', () => {
  it('renders title and message correctly', () => {
    const wrapper = mount(EmptyList);
    expect(wrapper.text()).toContain('Uh-oh!');
    expect(wrapper.text()).toContain('You look lost on your journey!');
  });

  it('renders BtnAtom with correct props', () => {
    const wrapper = mount(EmptyList);
    const button = wrapper.findComponent(BtnAtom);

    expect(button.exists()).toBe(true);
    expect(button.props('name')).toBe('Go back home');
    expect(button.props('widthBt')).toBe('155');
    expect(button.props('active')).toBe(true);
  });

  it('emits "manage:goBack" event when BtnAtom is clicked', async () => {
    const wrapper = mount(EmptyList);

    const button = wrapper.findComponent(BtnAtom);
    await button.vm.$emit('manage:click');

    expect(wrapper.emitted('manage:goBack')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const wrapper = mount(EmptyList);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
