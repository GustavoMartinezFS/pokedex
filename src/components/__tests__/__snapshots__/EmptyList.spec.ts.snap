// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmptyList.vue > matches snapshot 1`] = `
"<div data-v-8934a6a3="" class="empty-list">
  <h1 data-v-8934a6a3="">Uh-oh!</h1>
  <p data-v-8934a6a3="">You look lost on your journey!</p><button data-v-94a85ee2="" data-v-8934a6a3="" class="btn btn--no-icon btn--#F22539" style="max-width: 275px; min-width: 131px; height: 44px; padding: 10px; margin: 0px; background-color: rgb(242, 37, 57);">
    <!--v-if--><span data-v-94a85ee2="" class="text">Go back home</span>
  </button>
</div>"
`;
